src/Domain/MaxNumberOfNeighbors.hpp

```diff
@@ -23,3 +23,22 @@ constexpr size_t maximum_number_of_neighbors(const size_t dim) {
       throw "Invalid dim specified";
   };
 }
+
+/// \ingroup ComputationalDomainGroup
+/// Returns the maximum number of neighbors in each direction an element can
+/// have in `dim` dimensions.
+///
+/// \note Assumes a maximum 2-to-1 refinement between two adjacent Elements.
+constexpr size_t maximum_number_of_neighbors_per_direction(const size_t dim) {
+  switch (dim) {
```

Comment:
Why not `two_to_the(dim - 1)`?

Reply:
How would you suggest checking for invalid `dim`? Would you omit the check?

Reply:
Is this function called by anything that doesn't know the dimension at compile time?

Reply:
In this PR it is only used with a compile-time-known dimension.

Reply:
I don't see why anything other than 0 should be invalid.  Handling 0 separately might be worth it, either as a `throw` or just `return 0`.  (`two_to_the((size_t)-1)` is UB.)

Reply:
I've made this change.